pkgs/development/libraries/libmtp/default.nix

```diff
@@ -24,30 +31,26 @@ stdenv.mkDerivation rec {
     pkg-config
   ];
 
-  buildInputs = [
-    libiconv
-  ];
+  buildInputs = [ libiconv ];
 
-  propagatedBuildInputs = [
-    libusb1
-  ];
+  propagatedBuildInputs = [ libusb1 ];
+
+  preConfigure = "./autogen.sh";
 
-  preConfigure = ''
-    ./autogen.sh
-  '';
+  configureFlags = [ "--with-udev=${placeholder "out"}/lib/udev" ];
```

Comment:
Apparently this change moved the udev rules for libmtp from the `bin` to the `out` output, but `nixos/modules/services/desktops/gvfs.nix` still expects these rules to be in the `bin` output:
https://github.com/NixOS/nixpkgs/blob/fcb125441fdefeeea3e3f0a6dce725e7bce44fc7/nixos/modules/services/desktops/gvfs.nix#L57
Therefore the udev rules for libmtp do not get applied, which breaks access to MTP devices for users.

Should the udev rules be returned to the `bin` output, or should the `gvfs` module be adjusted to refer to `pkgs.libmtp.out` instead? (Note that plain `pkgs.libmtp` also refers to the `bin` output and therefore does not work in `services.udev.packages` at the moment.)

This problem also affects the NixOS 21.11 release.

Reply:
Addressed in #152289, nice catch!